tests/integration/adapter/incremental/test_schema_change.py

```diff
@@ -4,68 +4,183 @@
+            "schema_change_distributed_ignore.sql": schema_change_sql % ("distributed_incremental", "ignore"),
+            "schema_change_distributed_fail.sql": schema_change_sql % ("distributed_incremental", "fail"),
+            "schema_change_distributed_append.sql": schema_change_sql % ("distributed_incremental", "append_new_columns"),
         }
 
-    def test_ignore(self, project):
-        run_dbt(["run", "--select", "schema_change_ignore"])
-        result = project.run_sql("select * from schema_change_ignore order by col_1", fetch="all")
+    @pytest.mark.parametrize("model", ("schema_change_ignore", "schema_change_distributed_ignore"))
+    def test_ignore(self, project, model):
+        run_dbt(["run", "--select", model])
+        result = project.run_sql(f"select * from {model} order by col_1", fetch="all")
         assert len(result) == 3
         assert result[0][1] == 1
-        run_dbt(["run", "--select", "schema_change_ignore"])
-        result = project.run_sql("select * from schema_change_ignore", fetch="all")
+        run_dbt(["run", "--select", model])
+        result = project.run_sql(f"select * from {model}", fetch="all")
         assert len(result) == 5
 
-    def test_fail(self, project):
-        run_dbt(["run", "--select", "schema_change_fail"])
-        result = project.run_sql("select * from schema_change_fail order by col_1", fetch="all")
+    @pytest.mark.parametrize("model", ("schema_change_fail", "schema_change_distributed_fail"))
+    def test_fail(self, project, model):
+        run_dbt(["run", "--select", model])
+        result = project.run_sql(f"select * from {model} order by col_1", fetch="all")
         assert len(result) == 3
         assert result[0][1] == 1
         _, log_output = run_dbt_and_capture(
             [
                 "run",
                 "--select",
-                "schema_change_fail",
+                model,
             ],
             expect_pass=False,
         )
         assert 'out of sync' in log_output.lower()
 
-    def test_append(self, project):
-        run_dbt(["run", "--select", "schema_change_append"])
-        result = project.run_sql("select * from schema_change_append order by col_1", fetch="all")
+    @pytest.mark.parametrize("model", ("schema_change_append", "schema_change_distributed_append"))
+    def test_append(self, project, model):
+        run_dbt(["run", "--select", model])
+        result = project.run_sql(f"select * from {model} order by col_1", fetch="all")
         assert len(result) == 3
         assert result[0][1] == 1
-        run_dbt(["--debug", "run", "--select", "schema_change_append"])
-        result = project.run_sql("select * from schema_change_append order by col_1", fetch="all")
+        run_dbt(["--debug", "run", "--select", model])
+        result = project.run_sql(f"select * from {model} order by col_1", fetch="all")
         assert result[0][2] == 0
         assert result[3][2] == 5
+
```

Comment:
Would you mind also adding a test to check the drop columns logic?

Reply:
added assertions for the dropped column 👍

Reply:
the drop column logic is now tested in TestComplexSchemaChange:test_sync. we are just asserting the resulting column count at the moment. I could extend it to check for existing column names as well